src/ledger-bridge-keyring.ts

```diff
@@ -0,0 +1,883 @@
+          isConnectionChangedResponse(data.payload)
+        ) {
+          this.isDeviceConnected = data.payload.connected;
+        }
+      }
+
+      return undefined;
+    };
+    window.addEventListener('message', this.#eventListener);
+  }
+
+  destroy() {
+    if (this.#eventListener) {
+      window.removeEventListener('message', this.#eventListener);
+    }
+  }
+
+  async #getPage(increment: number) {
+    this.page += increment;
+
+    if (this.page <= 0) {
+      this.page = 1;
+    }
+    const from = (this.page - 1) * this.perPage;
+    const to = from + this.perPage;
+
+    await this.unlock();
+    let accounts;
+    if (this.#isLedgerLiveHdPath()) {
+      accounts = await this.#getAccountsBIP44(from, to);
+    } else {
+      accounts = this.#getAccountsLegacy(from, to);
+    }
+    return accounts;
+  }
+
+  async #getAccountsBIP44(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const path = this.#getPathForIndex(i);
+      const address = await this.unlock(path);
+      const valid = this.implementFullBIP44
+        ? await this.#hasPreviousTransactions(address)
+        : true;
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+
+      // PER BIP44
+      // "Software should prevent a creation of an account if
+      // a previous account does not have a transaction history
+      // (meaning none of its addresses have been used before)."
+      if (!valid) {
+        break;
+      }
+    }
+    return accounts;
+  }
+
+  #getAccountsLegacy(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const address = this.#addressFromIndex(pathBase, i);
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+      this.paths[ethUtil.toChecksumAddress(address)] = i;
+    }
+    return accounts;
+  }
+
+  // eslint-disable-next-line no-shadow
+  #addressFromIndex(basePath: string, i: number) {
+    const dkey = this.hdk.derive(`${basePath}/${i}`);
+    const address = ethUtil
+      .publicToAddress(dkey.publicKey, true)
+      .toString('hex');
+    return ethUtil.toChecksumAddress(`0x${address}`);
+  }
+
+  #pathFromAddress(address: string) {
+    const checksummedAddress = ethUtil.toChecksumAddress(address);
+    let index = this.paths[checksummedAddress];
+    if (typeof index === 'undefined') {
+      for (let i = 0; i < MAX_INDEX; i++) {
+        if (checksummedAddress === this.#addressFromIndex(pathBase, i)) {
+          index = i;
+          break;
+        }
+      }
+    }
+
+    if (typeof index === 'undefined') {
+      throw new Error('Unknown address');
+    }
+    return this.#getPathForIndex(index);
+  }
+
```

Comment:
Ah, looks like you removed `_toAscii` because it was never getting called, is that correct?

Reply:
Exactly! along with `#normalize` and `#paddLeftEven`. I'm adding a note to the PR description